packages/react-web/src/inbox/adapters/SignerAdapter.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  PendingSigningRequestError,
+  Signature,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import { SignArbitraryMessage } from '@lens-protocol/domain/use-cases/inbox';
+import { ActiveWallet } from '@lens-protocol/domain/use-cases/wallets';
+import { PromiseResultPresenter } from '@lens-protocol/react';
+import { Signer } from '@xmtp/react-sdk';
+
+export class SignerAdapter implements Signer {
+  constructor(private activeWallet: ActiveWallet) {}
+
+  async getAddress() {
+    const wallet = await this.activeWallet.requireActiveWallet();
```

Comment:
I would leave a comment here that highlight this being a shortcut and we would not normally have use cases returning data synchronously with their return statement but we would have decoupled the delivery mechanism via a presenter interface.

Reply:
hm, but this is not a use-case, this is a Xmtp Signer adapter, in the method below `signMessage` there is a usecase with a presenter (using PromiseResultPresenter)